### PR TITLE
Initial implementation of infinite scroller.

### PR DESCRIPTION
**Pull request by flackr**
Creates an infinitely scrollable region which is given data that is randomly generated with a delay to simulate network round trip times.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -24,8 +24,183 @@
...
+
+ContentSource.prototype = {
+  fetch: function(count) {
+    // Fetch at least 30 or count more objects for display.
+    count = Math.max(30, count);
+    return new Promise(function(resolve, reject) {
+      // Assume 50 ms per item.
+      setTimeout(function() {
+        var items = [];
+        for (var i = 0; i < Math.abs(count); i++) {
```
Comment (surma) [inline src/index.html]: I totally don’t get the `Math.abs` :smile:

File: src/scripts/infinite-scroll.js [hunk 2]
```diff
@@ -15,4 +15,397 @@
...
+      if (this.items_.length <= this.loadedItems_)
+        this.addItem_();
+      this.items_[this.loadedItems_++].data = items[i];
+    }
+    this.attachContent();
+  }
+}
+
+// Minimal Promise implementation for browsers which do not have promises.
+window.Promise = window.Promise || function(exec) {
```
Comment (surma) [inline src/scripts/infinite-scroll.js]: I think I’d remove this and have people pull in a proper polyfill if they want to run in a browser w/o Promises.

File: src/scripts/infinite-scroll.js [hunk 3]
```diff
@@ -15,4 +15,397 @@
...
+      return tombstone;
+    }
+    return this.source_.createTombstone();
+  },
+
+  /**
+   * Attaches content to the scroller and updates the scroll position if
+   * necessary.
+   */
+  attachContent: function() {
```
Comment (surma) [inline src/scripts/infinite-scroll.js]: This function is massive. I’ll see if I can do some refactoring here :)

**Comment by surma** [comment]
@flackr Pretty awesome. Had a few nits, but gonna merge anyways. I’ll see if I can make a refactoring pass over this in the near future.

File: src/scripts/infinite-scroll.js (hunk 2 above)
Comment (flackr) [inline src/scripts/infinite-scroll.js]: How about we move the polyfill to index.html or a separate JS file? My motivation is to be able to easily ensure this keeps working well on IE.

File: src/index.html (hunk 1 above)
Comment (flackr) [inline src/index.html]: You're totally right, the abs is a relic from a time when I wanted to be able to fetch new items in either forward or reverse directions from some existing items. We may still want to do something like this in order to link into the middle of a long scroller but I didn't want that to block getting something up.

File: src/scripts/infinite-scroll.js (hunk 2 above)
Comment (surma) [inline src/scripts/infinite-scroll.js]: Agreed. I’ll pull something in.

**Pull request merged**